pkg/utils/utils.go

```diff
@@ -49,3 +56,21 @@ func ValidateNetworkName(networkName string) *types.Error {
 	}
 	return nil
 }
+
+// ValidateInterfaceName will validate the interface name based on the three rules below
+// 1. The name must not be empty
+// 2. The name must be less than 16 characters
+// 3. The name must not contain / or any whitespace characters
+// ref to https://git.kernel.org/pub/scm/network/iproute2/iproute2.git/tree/lib/utils.c?id=1f420318bda3cc62156e89e1b56d60cc744b48ad#n827
+func ValidateInterfaceName(ifName string) *types.Error {
+	if len(ifName) == 0 {
+		return types.NewError(types.ErrInvalidEnvironmentVariables, "interface name is empty", "")
+	}
+	if len(ifName) > maxInterfaceNameLength {
+		return types.NewError(types.ErrInvalidEnvironmentVariables, "interface name is too long", fmt.Sprintf("interface name should be less than %d characters", maxInterfaceNameLength+1))
+	}
+	if bytes.ContainsAny([]byte(ifName), `/ `) {
```

Comment:
Sorry that I didn't caught that earlier, but what about new line, tab or vertical tab (I mean - rest of whitespaces)?
Could use there https://golang.org/pkg/unicode/#White_Space ?

Reply:
@jellonek Thanks for your suggestion, updated!